backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/descriptor/AdvancedSearchBPartnerProcessor.java

```diff
@@ -24,25 +24,57 @@
 
 import de.metas.bpartner.BPartnerLocationId;
 import de.metas.bpartner.service.IBPartnerDAO;
+import de.metas.ui.web.view.SqlViewFactory;
+import de.metas.ui.web.view.descriptor.SqlViewKeyColumnNamesMap;
+import de.metas.ui.web.window.datatypes.DocumentId;
+import de.metas.ui.web.window.datatypes.WindowId;
 import de.metas.ui.web.window.model.Document;
+import de.metas.user.UserId;
 import de.metas.util.Services;
 import lombok.NonNull;
 import org.adempiere.exceptions.AdempiereException;
+import org.compiere.model.I_C_BPartner_Adv_Search_v;
 import org.compiere.model.I_C_Order;
 
+import java.util.List;
+import java.util.Objects;
+
 public class AdvancedSearchBPartnerProcessor implements AdvancedSearchDescriptor.AdvancedSearchSelectionProcessor
 {
+	private final SqlViewFactory sqlViewFactory;
+
+	public AdvancedSearchBPartnerProcessor(final SqlViewFactory sqlViewFactory)
+	{
+		this.sqlViewFactory = sqlViewFactory;
+	}
 
 	@Override
-	public void processSelection(final Document document, final String bpartnerFieldName, final String selectionIdStr)
+	public void processSelection(final WindowId windowId, final Document document, final String bpartnerFieldName, final String selectionIdStr)
```

Comment:
fine for me to introduce WindowId param,
but you could use document.getDocumentPath().getWindowId()

Reply:
Not really. windowId is the ID of the search assistant window, while document is the sales order document to be populated.
![image](https://user-images.githubusercontent.com/60300682/122564531-52a12c00-cffa-11eb-9d3c-0b61b5d22eb1.png)